### ISSUE TITLE
Add convention test for `GetAll*` convention

### ISSUE THREAD

**Issue by alfhenrik** [octokit/octokit.net#770]
I thought perhaps that creating a convention test for ensuring that methods that use the new Pagination functionality follow the `GetAll*` naming convention that is being implemented.

I created the following convention test to check that methods that return `Task<IReadOnlyList<T>>`  and that starts with `Get`

```
        [Theory]
        [MemberData("ClientInterfaces")]
        public void CheckReadOnlyListNamingConventions(Type clientInterface)
        {
            foreach (var method in clientInterface.GetMethodsOrdered())
            {
                var returnType = method.ReturnType.GetTypeInfo();
                if (returnType.TypeCategory != TypeCategory.ReadOnlyList ||
                    !method.Name.StartsWith("Get"))
                {
                    continue;
                }

                Assert.True(method.Name.StartsWith("GetAll"), string.Format("[{0}.{1}]: Method: {2}", clientInterface.Namespace, clientInterface.Name, method.Name));
            }
        }

        public static IEnumerable<object[]> ClientInterfaces()
        {
            return typeof(IEventsClient).Assembly.ExportedTypes.Where(TypeExtensions.IsClientInterface).Select(type => new[] { type })
                .Concat(typeof(IObservableEventsClient).Assembly.ExportedTypes.Where(TypeExtensions.IsClientInterface).Select(type => new[] { type }));
        }
```

If this would be of help I'd be happy to send it through as a PR (together with fixing up the existing methods that currently fail the test)

**Comment by shiftkey** [comment]
:thumbsup: might have some overlap with the convention test I added in #760 but I'm fine for them to co-exist for a while...

**Comment by shiftkey** [comment]
Fixed in #771
